lib/oneview-sdk/rest.rb

```diff
@@ -118,20 +118,26 @@ def rest_delete(path, options = {}, api_ver = @api_version)
     # Uploads a file to a specific uri
     # @param [String] file_path
     # @param [String] path The url path starting with "/"
-    # @param [Hash] body_params The params to append to body of http request. Default is {}.
-    # @option body_params [String] 'name' The name to show (when resource accepts a name)
+    # @param [Hash] options The options for the request. Default is {}.
+    # @option options [String] :body Hash to be converted into json and set as the request body
+    # @option options [String] :header Hash to be converted into json and set as the request header
+    # @option options [String] :file_name String that defines the new file name
     # @param [Integer] timeout The number of seconds to wait for completing the request. Default is 300.
     # @return [Hash] The parsed JSON body of response
-    def upload_file(file_path, path, body_params = {}, timeout = READ_TIMEOUT)
+    def upload_file(file_path, path, options = {}, timeout = READ_TIMEOUT)
       raise NotFound, "ERROR: File '#{file_path}' not found!" unless File.file?(file_path)
-      options = {
+      options = options.dup
```

Comment:
This is fine, but if you wanted to do: `options = Hash[options.map { |k, v| [k.to_s, v] }]`, you could simplify some of the code below and just access the keys with one type (string).